### PR TITLE
Add initial support for M65S

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Adds support for Mode's SixtyFive M65S PCB. This is the solderable version; two more hotswap versions M65HA and M65HI will be PR'd once final prototypes are manufactured.

The ``keyboard.c`` features an I2C-EEPROM code for VIA compatibility (developed with @tzarc ) with adjacent debugging messages; for the scaled-release, debugging is turned off, hence its commenting in ``board_init()``.

Both VIA and non-VIA firmwares were thoroughly prototyped.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mode/m65s/rules.mk [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+EEPROM_DRIVER = i2c
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE -DDEBUG_EEPROM_OUTPUT=TRUE
+STM32_BOOTLOADER_ADDRESS = 0x1FFF0000
+#VIA_ENABLE=yes
+LTO_ENABLE=yes
+
+LTO_ENABLE = no
```
Comment (drashna) [inline keyboards/mode/m65s/rules.mk:27]:
```suggestion
LTO_ENABLE = no
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mode/m65s/rules.mk (hunk 1 above)
Comment (Gondolindrim) [reply keyboards/mode/m65s/rules.mk]: Committed in f99896465fd572f62109ee2bb70681ad394ae88c

File: keyboards/mode/m65s/m65s.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        eeconfig_update_user(tmp.raw);
+        uint32_t value = eeconfig_read_user();
+        if (value != tmp.raw) {
+            dprint("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n");
+            dprint("!! EEPROM readback mismatch!\n");
+            dprint("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n");
+        }
+    }
+}
```
Comment (zvecr) [inline keyboards/mode/m65s/m65s.c:112]:
```suggestion
```

File: keyboards/mode/m65s/rules.mk [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+EEPROM_DRIVER = i2c
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE -DDEBUG_EEPROM_OUTPUT=TRUE
+STM32_BOOTLOADER_ADDRESS = 0x1FFF0000
+#VIA_ENABLE=yes
+LTO_ENABLE = no
+OPT = 0
+OPT_DEFS += -g
```
Comment (zvecr) [inline keyboards/mode/m65s/rules.mk:27]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by Jaicob** [comment]
@zvecr can you please give this another look, or assign a different review who is available to?

**Review by tzarc — APPROVED** [review]

**Pull request merged**